### PR TITLE
Refactor login page

### PR DESCRIPTION
**Pull request by anbraten**
As requested in https://github.com/go-gitea/gitea/pull/31504#issuecomment-2196196646. This PR refactor the login page:

Screenshot from 2024-07-04 19-23-10

Screenshot from 2024-07-04 19-23-21

Screenshot from 2024-06-30 09-35-20

# Changes
- [x] use separate box for passkey login and go to registration
- [x] move forgot passoword next to password label
- [x] fix password required label `*` and padding 
- [x] remove tabs from login page

### CODE REVIEW

**Comment by silverwind** [comment]
Looks sexy, one issue that I see is the spacing below the Password text is not equal to the one below Username, please make it use the same spacing as below Username:

image

**Comment by silverwind** [comment]
Oh and I would make the "Register Account" button also full width.

**Comment by anbraten** [comment]
> Looks sexy, one issue that I see is the spacing below the Password text is not equal to the one below Username, please make it use the same spacing as below Username:


Do you have an idea how to do this "nicely"? Problem was that the label is now nested in a div to be able to have the `forgot password` link at the right, however the required and padding is broken.

**Comment by silverwind** [comment]
Did some tweaks on signup, button is full width, captchas are centered:

image

**Comment by silverwind** [comment]
Not sure I like the logo on the signin page, it does take away a bit of vertical space and there is already a logo on the header.

image

**Comment by anbraten** [comment]
Thanks for the adjustments.

> Not sure I like the logo on the signin page, it does take away a bit of vertical space and there is already a logo on the header.

True, its not perfect yet. Without the logo it looked a bit too much like a dashboard card to me, but the header and the logo underneath isn't nice as well.

Could you check if we can somehow get the red required star for the password back.

**Comment by silverwind** [comment]
> Could you check if we can somehow get the red required star for the password back.

Hmm it looks like Fomantic UI's CSS expects the label as a direct descendant:

```css
.ui.form .required.field > label::after
```

So we likely need to remove the wrapper and find other solutions.

**Comment by silverwind** [comment]
But we also just can't add it to label as the `::after` would appear after the link:

image

This will need a bit of hacky CSS. Luckily this indicator is not interactive and seem to be a pure CSS feature.

**Comment by silverwind** [comment]
Found a solution with `display: grid`. It's the least hacky fix I could think of.

image

File: templates/user/auth/oauth_container.tmpl [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+			{{if .EnableOpenIDSignIn}}
+				<a class="openid ui button tw-flex tw-items-center tw-justify-center tw-py-2 tw-w-full" href="{{AppSubUrl}}/user/login/openid">
+				{{svg "fontawesome-openid" 28 "tw-mr-2"}}
+				{{ctx.Locale.Tr "sign_in_with_provider" "OpenID"}}
+				</a>
+			{{end}}
+		</div>
+	</div>
+</div>
+{{end}}
```
Comment (silverwind) [inline templates/user/auth/oauth_container.tmpl]:
```suggestion
{{end}}

```

**Comment by anbraten** [comment]
Removed the logo and did some smaller adjustments. PR should be ready from my side now.

**Comment by lunny** [comment]
> Removed the logo and did some smaller adjustments. PR should be ready from my side now.

Could you also update the screenshots?

**Comment by anbraten** [comment]
Sure updated them.

**Review by silverwind — APPROVED** [review]

**Review by denyskon — APPROVED** [review]
Looks nice 🚀

**Pull request merged**